src/SharedUtilities/implicit_tendencies.jl

```diff
@@ -0,0 +1,61 @@
+export make_tendency_jacobian,
+    make_update_jacobian, AbstractTridiagonalW, ∂tendencyBC∂Y
+
+
+"""
+   make_tendency_jacobian(model::AbstractModel)
+
+Creates and returns a function which updates the auxiliary
+variables `p` in place and then updates the entries of the 
+Jacobian matrix `W` for the `model` in place.
+
+The default is that no updates are required, no implicit tendency is
+present, and hence the timestepping is entirely explicit.
+"""
+function make_tendency_jacobian(model::AbstractModel)
+    update_aux! = make_update_aux(model)
+    update_jacobian! = make_update_jacobian(model)
+    function tendency_jacobian!(W, Y, p, dtγ, t)
+        update_aux!(p, Y, t)
+        update_jacobian!(W, Y, p, dtγ, t)
+    end
+    return tendency_jacobian
+end
+
+"""
+    make_update_jacobian(model::AbstractModel)
+
+Creates and returns a function which updates the entries
+of the Jacobian matrix `W` in place.
+
+If the implicit tendency function is given by 
+`T!(dY, Y, p, t) = make_implicit_tendency(model)`, the Jacobian 
+should be given by `W_{i,j}! = ∂T!_i/∂Y_j`, where `Y_j` is the 
+`j-th` state variable
+and `T!_i` is the implicit tendency of the `i-th` state variable.
+
+The default is that no updates are required, no implicit tendency is
+present, and hence the timestepping is entirely explicit.
+"""
+function make_update_jacobian(model::AbstractModel)
+    function update_jacobian!(W, Y, p, dtγ, t) end
+    return update_jacobian
+end
+
+"""
+    ∂tendencyBC∂Y(::AbstractModel, ::AbstractBC, ::AbstractBoundary, _...)
+
+A function stub which returns the derivative of the implicit tendency
+term of the `model` arising from the boundary condition,
+with respect to the state Y.
+"""
+function ∂tendencyBC∂Y(
```

Comment:
It seems like we could come up with a better name for this function - maybe `\partial T_BC \partial Y`? That might be worse though because `T_BC` sounds like temperature at the boundary. What do you think?

Reply:
I'm going to leave this alone for now, but I'm happy to revisit it as we continue work on implicit solvers! In particular, I think we'll have to change the function a little for cases with more than one variable.